gpytorch/mlls/exact_marginal_log_likelihood.py

```diff
@@ -32,12 +32,15 @@ def forward(self, output, target):
         res = output.log_prob(target)
 
         # Add terms for SGPR / when inducing points are learned
-        trace_diff = torch.zeros_like(res)
-        for variational_strategy in self.model.variational_strategies():
-            if isinstance(variational_strategy, MVNVariationalStrategy):
-                trace_diff = trace_diff.add(variational_strategy.trace_diff())
-        trace_diff = trace_diff / self.likelihood.log_noise.exp()
-        res = res.add(0.5, trace_diff)
+        added_loss = torch.zeros_like(res)
+        for added_loss_term in self.model.added_loss_terms():
+            if isinstance(added_loss_term, InducingPointKernelAddedLossTerm):
```

Comment:
This is really hacky. I think `InducingPointKernel` should take in the GaussianLikelihood in the constructor. This way, the added loss term can deal with the noise itself.

(I realize what we had before was also pretty hacky, but if we're fixing it we should do something a bit better)

Reply:
Will fix this, good point